src/libraries/Common/src/Interop/Unix/System.Native/Interop.IPAddress.cs

```diff
@@ -73,16 +73,9 @@ public bool Equals(IPAddress other)
 
                 fixed (byte* thisAddress = Address)
                 {
-                    for (int i = 0; i < addressByteCount; i++)
-                    {
-                        if (thisAddress[i] != other.Address[i])
-                        {
-                            return false;
-                        }
-                    }
+                    return new ReadOnlySpan<byte>(thisAddress, addressByteCount).SequenceEqual(
+                           new ReadOnlySpan<byte>(other.Address, addressByteCount));
```

Comment:
`other.Address` is `fixed byte Address[MAX_IP_ADDRESS_BYTES]`. How does this work?

Cam we get rid of `this.Address` pinning too?

Reply:
`other` is a local on the stack, so C# doesn't require it be pinned with fixed. `this` could be in an object on the heap, though.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAAQCuAdgM4CGAZjKdQC5TlgukDCphA3oVLDSAS0osYUSrQA2peqIQwAJqWABPSaQAitFrQDaOJAF0A3EJFXhJUhhSkAsgApeEFgAspAShulBAhFg9S0YACpSAAccUgBeXX1aSyCQ4U1JSKiseNIPbygAOj0DFOCAX0JyoA==